nautobot/docs/plugins/development.md

```diff
@@ -1146,3 +1146,147 @@ try:
+    view_mixins.ObjectEditViewMixin,
+    view_mixins.ObjectDestroyViewMixin,
+):
+
+    filterset_class = YourPluginModelFilterSet
+    filterset_form_class = YourPluginModelFilterForm
+    form_class = YourPluginModelForm
+    queryset = YourPluginModel.objects.all()
+    serializer_class = serializers.YourPluginModelSerializer
+    table_class = YourPluginModelTable
+    # You do not need to specify attributes that are not needed.
+```
+
+### Template Naming for NautobotUIViewSet
+
+Template naming is very intuitive in NautobotUIViewSet. In `templates/yourpluginmodel` folder, name your templates following this convention `{app_label}/{model_name}_{self.action}.html`.
+
+| ViewMixins    | self.action   |
+| ------------- |:-------------:|
+| ObjectlistViewMixin        | list         |
+| ObjectDetailViewMixin      | retrieve     |
+| ObjectEditViewMixin        | create/update|
+| ObjectDestroyViewMixin     | destroy      |
+| ObjectBulkDestroyViewMixin | bulk_destroy |
+| ObjectBulkCreateViewMixin  | bulk_create  |
+| ObjectBulkUpdateViewMixin  | bulk_update  |
+
+For example, for a DetailView template for `yourpluginmodel`, the template name will be `yourplugin/yourpluginmodel_retrieve.html`, for a BulkCreateView template for `yourpluginmodel`, the template name will be `yourplugin/yourpluginmodel_bulk_create.html` and etc.
+
+If you do not specify your own templates in `templates/yourpluginmodel` folder, `NautobotUIViewSet` will fall back to `generic/object_{self.action}.html`.
+
+If you do specify a `{app_label}/{model_opts.model_name}_create.html` file but not a `{app_label}/{model_opts.model_name}_update.html` file. When you update an object, it will fall back to `{app_label}/{model_opts.model_name}_create.html` and vice versa.
+
+## NautobotUIViewSetRouter
+
+With `NautobotUIViewSet` as the base UI ViewSet for `YourPluginModel`, it is required to register your urls with the help of `NautobotUIViewSetRouter`.
+
+For a concrete example on how to use `NautobotUIViewSetRouter`, see `nautobot.circuits.urls`.
+
+Below is a theoretical `urls.py` file for `YourPluginModel`:
+
+```python
+from django.urls import path
+
+from nautobot.core.views.routers import NautobotUIViewSetRouter
+from nautobot.plugins import views
+
+
+router = NautobotUIViewSetRouter()
+router.register("yourpluginmodel", views.YourPluginModelUIViewSet)
+
+urlpatterns = [
+    # Extra urls that do not follow the patterns of `NautobotUIViewSetRouter` go here.
+    # changelog, notes and etc.
+    ...
+    path(
+        "yourpluginmodels/<slug:slug>/changelog/",
+        ObjectChangeLogView.as_view(),
+        name="yourpluginmodel_changelog",
+        kwargs={"model": yourpluginmodel},
+    ),
+    path(
+        "yourpluginmodels/<slug:slug>/notes/",
+        ObjectNotesView.as_view(),
+        name="yourpluginmodel_notes",
+        kwargs={"model": yourpluginmodel},
+    ),
+    ...
+]
+urlpatterns += router.urls
+```
+
+### Excluding Urls from NautobotUIViewSetRouter
```

Comment:
Should this be a separate section or can it just be an additional sentence or two under `Excluding ViewMixins from NautobotUIViewSet`?

Reply:
I can move it under `Excluding ViewMixins from NautobotUIViewSet`